drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/rightpanel/TestToolsPresenter.java

```diff
@@ -359,7 +363,9 @@ public void onDisableEditorTab() {
     public void setSelectedElement(ListGroupItemView selected) {
         selectedListGroupItemView = selected;
         selectedFieldItemView = null;
-        if (filterTerm(selected.getFactName(), listGroupItemPresenter.getFilterTerm(), false)) {
+        boolean isDMNScenario = (Objects.equals(modelType, ScenarioSimulationModel.Type.DMN));
+        boolean toBeDisabled = selected.isInstanceAssigned() && (isDMNScenario || isSimple(selected.getFactName())) ;
+        if (toBeDisabled || filterTerm(selected.getFactName(), listGroupItemPresenter.getFilterTerm(), selected.isInstanceAssigned())) {
```

Comment:
Can you please add a comment and/or wrap this logic in a method to improve the readability?

Reply:
Done